aten/src/ATen/native/SharedReduceOps.h

```diff
@@ -21,9 +21,26 @@
 #define device_sqrt std::sqrt
 #endif
 #if defined(__CUDACC__) || defined(__HIPCC__)
-#define MAX(X, Y) ::max(X,Y)
-#define MIN(X, Y) ::min(X,Y)
+template <typename scalar_t>
+inline C10_DEVICE scalar_t max_propagate_nan(scalar_t a, scalar_t b) {
+  if (at::_isnan(a) || at::_isnan(b)) {
+    return std::numeric_limits<scalar_t>::quiet_NaN();
+  } else {
+    return ::max(a, b);
+  }
+}
+template <typename scalar_t>
+inline C10_DEVICE scalar_t min_propagate_nan(scalar_t a, scalar_t b) {
+  if (at::_isnan(a) || at::_isnan(b)) {
+    return std::numeric_limits<scalar_t>::quiet_NaN();
+  } else {
+    return ::min(a, b);
+  }
+}
+#define MAX(X, Y) max_propagate_nan(X,Y)
```

Comment:
I'm not sure if this is the best way to add nan propagation. Maybe there's something more standard?

Reply:
@ngimel and I like this approach - is there an updated test for torch.norm nan propagation?

Reply:
Yep, I added one